pkg/querier/tail.go

```diff
@@ -103,22 +104,31 @@ func (t *Tailer) readTailClients() {
 // keeps sending oldest entry to responseChan. If channel is blocked drop the entry
 // When channel is unblocked, send details of dropped entries with current entry
 func (t *Tailer) loop() {
-	ticker := time.NewTicker(checkConnectionsWithIngestersPeriod)
-	defer ticker.Stop()
+	checkConnectionTicker := time.NewTicker(checkConnectionsWithIngestersPeriod)
+	defer checkConnectionTicker.Stop()
+
+	tailMaxDurationTicker := time.NewTicker(t.tailMaxDuration)
+	defer tailMaxDurationTicker.Stop()
 
 	tailResponse := new(TailResponse)
 
 	for {
 		if t.stopped {
-			break
+			return
 		}
 
 		select {
-		case <-ticker.C:
+		case <-checkConnectionTicker.C:
 			// Try to reconnect dropped ingesters and connect to new ingesters
 			if err := t.checkIngesterConnections(); err != nil {
 				level.Error(util.Logger).Log("Error reconnecting to disconnected ingesters", fmt.Sprintf("%v", err))
 			}
+		case <-tailMaxDurationTicker.C:
+			if err := t.close(); err != nil {
+				level.Error(util.Logger).Log("Error closing Tailer", fmt.Sprintf("%v", err))
+			}
+			t.closeErrChan <- errors.New("Reached tail max duration limit")
```

Comment:
Error strings should not start with a capital letter because they'll often be prefixed before printing.

see https://github.com/golang/go/wiki/Errors